### PR TITLE
Add macros for known names and paths

### PR DESCRIPTION
**Pull request by flodiebold**

### CODE REVIEW

**Comment by matklad** [comment]
Looking at the diff, name! and path! would be more readable and not much more verbose

File: crates/ra_hir_def/src/path.rs [hunk 1]
```diff
@@ -297,68 +297,63 @@ impl From<Name> for Path {
...
-        Path::from_simple_segments(
-            PathKind::Abs,
-            vec![name::STD, name::ITER, name::INTO_ITERATOR_TYPE],
-        )
+        P![std::iter::IntoIterator]
     }
 
     pub fn std_ops_try() -> Path {
-        Path::from_simple_segments(PathKind::Abs, vec![name::STD, name::OPS, name::TRY_TYPE])
+        P![std::ops::Try]
```
Comment (matklad) [inline crates/ra_hir_def/src/path.rs]: Can we make P![std::ops::try] the public API? We probably still should check for a whitelist of paths though

Comment (flodiebold) [reply crates/ra_hir_def/src/path.rs]: Hmm you're right we could just do that... how would you do the path whitelist though?

Comment (matklad) [reply crates/ra_hir_def/src/path.rs]: By creating a macro rules branch for each allowed path? Or passing paths through an id macro, which is partially defined

**Comment by flodiebold** [comment]
@matklad How about this?

**Pull request merged**